client/run.go

```diff
@@ -290,6 +303,67 @@ func (cli *HyperClient) JsonFromCmdline(cmdArgs, cmdEnvs, cmdPortmaps []string,
 	return string(jsonString), nil
 }
 
+func parseVolume(volStr string) (*pod.UserVolume, *pod.UserVolumeReference, error) {
```

Comment:
speed and resouce are unimportant here.
And in golang, pointer for small object is slower and uses more resource, golang always allocate heap memory for pointers.